funnel/templates/event.html

```diff
@@ -46,37 +46,251 @@
+              //Participant has been checked-in so remove from 'checkin-queue' 
+              window.Queue.dequeue("checkin-queue", participantID);
+            }
+            else {
+              $(participant).siblings('.loader').show();
+            }
+          }
+          else {
+            if($(participant).hasClass('js-checkin')) {
+              //Participant's check-in has been cancelled so remove from 'cancelcheckin-queue' 
+              window.Queue.dequeue("cancelcheckin-queue", participantID);
+            }
+            else {
+              $(participant).siblings('.loader').show();
+            }
+          }
+        });
+      }
+    }
+
+    function getParticipantsData() {
+      $.ajax({
+        type: 'GET',
+        url:  '{{ "{0}event/{1}/participants/json".format(space.url_for(), event.name) }}',
+        timeout: 5000,
+        dataType: 'jsonp',
+        success: function(data) {
+          participantTableContent.set('participants', data.participants).then(function() {
+            if (Modernizr.localstorage) {
+              $('.loader').hide();
+              updateQueue("checkin-queue");
+              updateQueue("cancelcheckin-queue");
+            }
+          });
+        }
+      });
+    }
+
+    function postCheckinStatus(participantID, action) {
+      if(action) {
+        var posturl = "{{ '{0}event/{1}/checkin/'.format(space.url_for(), event.name) }}" + participantID + "?checkin=t";
+      }
+      else {
+        var posturl = "{{ '{0}event/{1}/checkin/'.format(space.url_for(), event.name) }}" + participantID;
+      }
+      $.ajax({
+        type: 'post',
+        url:  posturl,
+        timeout: 5000,
+        dataType: "json",
+        success: function(data) {
+          if(data.checked_in) {
+            window.Queue.dequeue("checkin-queue", data.participant_id);
+          }
+          else {
+            window.Queue.dequeue("cancelcheckin-queue", data.participant_id);
+          }
+        }
+      });
+    }
+
+    function checkQueues() {
+      //Read first participant from 'checkin-queue' and post check-in to server
+      var participantID = window.Queue.peek("checkin-queue");
+      if(participantID) {
+        postCheckinStatus(participantID, true);
+      }
+      //Read first participant from 'cancelcheckin-queue' and post cancel check-in to server
+      participantID = window.Queue.peek("cancelcheckin-queue");
+      if(participantID) {
+        postCheckinStatus(participantID, false);
+      }
+    }
+
+    //Get the list of participants from server
+    getParticipantsData();
+
+    //If localStorage is available, register event handlers for check-in, cancel-checkin and abort
+    if (Modernizr.localstorage) {
+      $('#participants-table-content').on('click', '.js-checkin', function(event) {
+        event.preventDefault();
+        //Add to participant_id to "checkin-queue"
+        window.Queue.enqueue("checkin-queue", $(this).data('pid'));
```

Comment:
So, what's happening here is, `Queue` has no knowledge about `checkin-queue`, since it's just passing it off to `localStorage`. I would recommend an object-oriented approach here. 
```javascript
   var checkinq = new Queue('checkin');
   checkinq.enqueue(item);
   checkinq.dequeue(item);
````
````